client.js

```diff
@@ -45,3 +47,22 @@ document.addEventListener('DOMContentLoaded', function () {
     return new EditorToolbar(dom.find('[' + references.referenceAttribute + '*="/components/clay-kiln"]'));
   }
 });
+
+// handle connection loss
+
+window.addEventListener('load', function () {
+  // test connection loss on page load
+  if (!navigator.onLine) {
+    // we're offline!
+    progress.open('grey', connectionLostMessage);
+  }
+});
+
+window.addEventListener('online', function () {
+  progress.close(); // in case there are any status messages open, close them
+});
+
+window.addEventListener('offline', function () {
+  progress.done('grey'); // turn any progress indicators to grey and end them
+  progress.open('grey', connectionLostMessage);
+});
```

Comment:
Things should be responsible to drawing themselves.

Start('publish') would make more sense.